### ISSUE TITLE
Warn against redundant T.let usage

### ISSUE THREAD

**Issue by ptarjan** [sorbet/sorbet#2255]
### Problem

It would be great if we could auto-detect the use of `T.let `when the type of the expression is the same as that to which it is being assigned i.e. redundant usage of `T.let`.

### Proposed solution

Add an autocorrect to remove these

**Comment by jez** [comment]
I disagree that we should have this autocorrect.

1. They're needed for type annotations for constants / instance variables
2. Type annotations document programmer intent. If the `T.let` is redundant, then the point of it is to prevent against the type changing to something else.

**Comment by atty-stripe** [comment]
I agree that in cases where `T.let` is used to assert the type, it displays programmer intent and should remain as-is.

However, it's also used today to assign types to variables that would otherwise be `T.untyped`. Later, if the underlying code has been updated to be typed, the `T.let` is redundant. In this case, it is preferable to remove the `T.let` usage.

**Comment by Morriar** [comment]
I like @ptarjan proposal, as the code evolves and more things are typed some of these `let` are not useful anymore and are just noise at this point.

While I agree with @jez that *some* annotations are required or can be used to mark the intent, some other are actually relics of the past and I would like an automated way to remove them.

From our study around Sorbet usage, what people like the least is when Sorbet can be seen inside the body of a method (all inline constructs like `let`, `cast`, `must`...) and I'm curious to know how many of them we can remove.

If you refuse this feature in default Sorbet, what do you think about adding this behind an option like `--remove-useless-let`?
